functions/completed-checks/v1.js

```diff
@@ -33,27 +33,27 @@ async function handleCompletedCheck (context, completedCheckMessage) {
   try {
     checkData = await sqlUtil.sqlFindCheckWithFormDataByCheckCode(completedCheckMessage.checkCode)
   } catch (error) {
-    context.log.error(`completed-check: ERROR: failed to retrieve checkData for checkCode: ${completedCheckMessage.checkCode}`)
+    context.log.error(error)
     throw error
   }
 
-  if (!canCompleteCheck(checkData.code)) {
-    const errorMessage = `completed-check: ERROR: check ${completedCheckMessage.checkCode} is not in a correct state to be completed. Current state is ${checkData.code}`
+  if (!canCompleteCheck(checkData.checkStatusCode)) {
+    const errorMessage = `completed-check: ERROR: check ${completedCheckMessage.checkCode} is not in a correct state to be completed. Current state is ${checkData.checkStatusCode}`
```

Comment:
have you considered what happens to this message after you have thrown it? as far as i can tell, it will go back on the queue to be picked up again until it reaches the limit and ends up on the poison queue. Is this the best way to deal with it?

Reply:
@GuyHarwood @jon-shipley as discussed we are going to tackle this separately.